src/pack_kernel.cpp

```diff
@@ -254,18 +269,21 @@ void clover_unpack_message_top(int x_min, int x_max, int y_min, int y_max,
 		// DO j=x_min-depth,x_max+x_inc+depth
 
 
-		_Pragma("kernel1d")
+		double *field = field_buffer.data;
+		const int field_sizex = field_buffer.sizeX;
+		double *top_rcv = top_rcv_buffer.data;
+		#pragma omp target teams distribute parallel for simd omp_use_target(use_target)
```

Comment:
I'm not sure this will work as we expect with MPI. One of these arrays will have come from an MPI buffer, so will be host-only and needs mapping to the device at the point of unpack.
Pack will similarly need to map the data.

Reply:
Fixed in https://github.com/UoB-HPC/cloverleaf_openmp_target/pull/2/commits/37d197ade0f4ff330fd969c20502a7a86100139f